suggested-content/word-2-vec-service.py

```diff
@@ -0,0 +1,63 @@
+import gensim, logging, sys
+from flask import Flask, request, jsonify
+from flask.ext.cors import CORS
+
+app = Flask(__name__)
+CORS(app, resources=r'/*',
+     allow_headers='*')
+
+MODEL = None
+
+## Logger magic
+logging.basicConfig(format='%(asctime)s : %(levelname)s : %(message)s', level=logging.INFO)
+
+
+@app.route('/check-phrases', methods=['POST'])
+def check():
+    phrases = request.json["phrases"]
+    vector = map(checkProximity, phrases)
+    transformed = map(lambda xy: {'name': xy[0], 'distance': xy[1]},
+                      [val for sublist in vector for val in sublist])
+
+    return jsonify(results = transformed)
+
+
+@app.route('/query', methods=['GET'])
+def query():
+    searchword = request.args.get('word', '')
+    notwords = request.args.get('not', '')
+    vector = checkProximity(searchword, notwords)
+    return jsonify(vector)
+
+def loadModel(filename):
+    return gensim.models.Word2Vec.load_word2vec_format(filename, binary=True)
+
+def checkProximity(phrase, notwords = None):
+    if notwords is not None:
+        try:
+            result = MODEL.most_similar(positive=phrase.split(),negative=notwords.split())
+            return result
+        except KeyError:
+            return []
+    else:
+        try:
+            result = MODEL.most_similar(positive=phrase)
+            return result
+        except KeyError:
+            return []
+
+
+def setup(argv):
+    if len(argv) != 2:
+        print("./word-2-vec-service <training-set>")
+        sys.exit(1)
+
+        global MODEL
```

Comment:
I'm am always suspicious of global and you don't need it here as MODEL is a package level variable.

See also app.config and app.config.from_object which seems more elegant for your usecase